test/integrationtests/voight_kampff/features/steps/utterance_responses.py

```diff
@@ -98,7 +117,6 @@ def _match_dialog_patterns(dialogs, sentence):
     dialogs = [re.sub(r'{.*?\}', r'.*', dia) for dia in dialogs]
     # Remove left over '}'
     dialogs = [re.sub(r'\}', r'', dia) for dia in dialogs]
-    dialogs = [re.sub(r' .* ', r' .*', dia) for dia in dialogs]
```

Comment:
ok i think this is actually a bug. 

my dialog file:

```
(OK | ) turning on the TV
```

My test file:

```
      Matching with the dialog file: None
Feature: turn-on-tv
  Scenario: Turning on the TV
    Given an English speaking user
    When the user says "turn on the tv"
    Then "tv-remote-control-skill" should reply with "turning on the tv"
```

When I test with this, I get

```
Dialogs4: ['^(ok | ) turning on the tv']
```

after this line. I think it should probably be something like `'.* turning on the tv'`, right?

I thought I had this working locally but don't anymore. Will keep trying

Reply:
or maybe this never worked?

Reply:
oh I put it in the wrong order. this is what works

code:

```
    # Allow custom fields to be anything
    dialogs = [re.sub(r'{.*?\}', r'.*', dia) for dia in dialogs]
    print("Dialogs1: {}".format(dialogs))
    # Remove left over '}'
    dialogs = [re.sub(r'\}', r'', dia) for dia in dialogs]
    print("Dialogs2: {}".format(dialogs))
    dialogs = [re.sub(r' [^\)]* ', r' .*', dia) for dia in dialogs]
    print("Dialogs3: {}".format(dialogs))
    # Merge consequtive .*'s into a single .*
    dialogs = [re.sub(r'\.\*( \.\*)+', r'.*', dia) for dia in dialogs]
    print("Dialogs4: {}".format(dialogs))
    # Remove double whitespaces
    dialogs = ['^' + ' '.join(dia.split()) for dia in dialogs]
    print("Dialogs5: {}".format(dialogs))
```

output:

```
      Dialogs1: ['(ok | ) turning on the tv']
      Dialogs2: ['(ok | ) turning on the tv']
      Dialogs3: ['(ok .*) .*tv']
      Dialogs4: ['(ok .*) .*tv']
      Dialogs5: ['^(ok .*) .*tv']
```

Reply:
I'd argue though at this point the regex might be harming more than it's helping

1. Hard to understand/maintain
2. Seems to have bugs e.g. it's stripping `turning on the` from my dialog. 

Might be better to expand this to a set of phrases and then just match with a simpler regex. Or is there a way to leverage the code that actually parses these files? If you could share a module between this code and that code it would guarantee that updates to the dialog syntax are always reflected in the test behaviour.

Reply:
I didn't realize the parenthesis expansion was run on the dialogfiles (or maybe I've forgotten), and yes it should be expanded using the normal parenthesis expansion. The regex intention is to handle the data entries:
`hello dear {person_type}`. and replace the `{person_type}` with `.*` since that could be anything. There doesn't really exist any dialog matching system anywhere since it's not an input to Mycroft.

Will try to improve it.

Reply:
Hmm, and to further improve this I think we can may be able to use normal dialog renderer to render the final regex in one step...

Reply:
I did a small change to expand parentheses in the dialog lines, which made your example work for me(TM). I will attempt a bigger change tonight to try to remove the big weird multi-pass regex

Reply:
I did another pass at it using the normal dialog loader to remove some of the regex clutter together with using the built in string format-parser...

Need to test it some more though

Reply:
Love the look of the new approach. It seems to be working now but check my other comment